### PR TITLE
remove old behavior on requestobject on request refactor

### PR DESCRIPTION
**Pull request by stafyniaksacha**
Changed the way that requestObject is builded.

seen with @ballinette 
We need to organize a workshop to speak about refactoring some things on all inputs in kuzzle (there is a lot of divergences)

### CODE REVIEW

**Comment by ballinette** [comment]
see also https://github.com/kuzzleio/kuzzle/issues/342

**Comment by dbengsch** [comment]
+1

File: models/requestObject.js [hunk 1]
```diff
@@ -59,10 +59,7 @@ function construct (object, additionalData, protocol) {
     this.data.body = object.body;
   }
   else if (object.query !== undefined) {
-    this.data = object;
-  }
-  else if (additionalData.query !== undefined) {
-    this.data = additionalData;
+    this.data.body = {query: object.query};
```
Comment (scottinet) [inline models/requestObject.js]: I don't remember where the `additionalData` can be used, but wouldn't it be safer to keep the test `else if (additionalData.query...` and simply store the query part of it in `this.data.body`?

**Comment by scottinet** [comment]
+1

**Comment by ballinette** [comment]
+1

**Pull request merged**